django_lightweight_queue/views.py

```diff
@@ -0,0 +1,78 @@
+import io
+import json
+import logging
+
+from django import template
+from django.db import transaction
+from django.conf import settings
+from django.http import Http404, HttpResponse
+
+from .job import Job
+
+
+# Note: no authentication here -- ensure you only connect up this view in
+# development!
+def debug_run(request):
+    """
+    Run a task on a `GET` request and within a transaction which will always be
+    rolled back.
+
+    This is useful for debugging tasks and for optimising their database
+    accesses (via the Django Debug Toolbar).
+
+    Logging output from the task is captured and included on the page, though
+    `print` calls are not.
+
+    You should take care when using this to debug tasks which interact with
+    services other than the database (e.g: caches or HTTP endpoints), as *only*
+    the database transaction is rolled back after the task completes. Any other
+    external requests will happen as normal.
+
+    To make this view available you'll need to have your main Django project
+    include its url somewhere. You are *strongly* encouraged only to do this in
+    DEBUG mode:
+    ```
+    if settings.DEBUG:
+        urlpatterns += (
+            url(r'', include('django_lightweight_queue.urls', namespace='django-lightweight-queue')),
+        )
+    ```
+    """
+
+    if not settings.DEBUG:
+        raise Http404("Debug view only available when DEBUG=True")
+
+    job = Job.from_json(request.GET['job'])
+
+    log_steam = io.StringIO()
+    handler = logging.StreamHandler(log_steam)
+
+    try:
+        logging.root.addHandler(handler)
+        with transaction.atomic():
+            result = job.run()
```

Comment:
What happens if `job.run()` raises an exception? I assume regular django debug mode error processing takes over and you get a nice stack trace?

Reply:
As far as I know `job.run()` doesn't raise exceptions -- it catches `Exception` internally and the returned boolean signifies whether the underlying function completed without error.